### PR TITLE
Introduce database migration tool

### PR DESCRIPTION
**Pull request by Lekuruu**
Due to some new people that want to contribute to the project, there has been demand for a proper database migration system, since it's tedious and annoying to always keep the database up-to-date by hand.

@ascenttree suggested to use [golang-migrate](https://github.com/golang-migrate/migrate), which should allow you to do this job very easily in theory. I still have to explore this tool and see if there are any upsides and downsides to it.

### CODE REVIEW

**Comment by Lekuruu** [comment]
@ascenttree Thoughts?

File: SETUP.md [hunk 1]
```diff
@@ -10,7 +10,33 @@
 
 ### Applying migrations
 
-Log in to your postgres server with your database management tool of choice, and apply/run this [base.sql](https://github.com/osuTitanic/titanic/blob/main/migrations/base.sql) file, that contains all the tables needed.
+Follow the installation instructions for [golang-migrate](https://github.com/golang-migrate/migrate/blob/master/cmd/migrate/README.md#installation).
```
Comment (ghost) [inline SETUP.md]: I think this should be good? 
```suggestion
Follow the installation guide for [golang-migrate](https://github.com/golang-migrate/migrate/blob/master/cmd/migrate/README.md#installation).
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: SETUP.md (hunk 1 above)
Comment (Lekuruu) [reply SETUP.md]: Sure why not

**Comment by Lekuruu** [comment]
image

Hmm.. I wonder where this came from. :trollface:

**Comment by ghost** [comment]
> image
> 
> Hmm.. I wonder where this came from. :trollface:

Jesus dear christ what is this

**Comment by Lekuruu** [comment]
> Jesus dear christ what is this

Its just columns, upon columns of beatmaps :D

**Comment by Digitalfear117** [comment]
What in the heck

**Pull request merged**